File: public/dsa/arrays/positive_cumulative_sum/optimal/positive_cumulative_sum.cpp
// imports-start
#include <bits/stdc++.h>
using namespace std;
// imports-end

vector<int> getPositiveCumulativeSum(vector<int> &arr) {
    int cumulative_sum = 0;
    vector<int> result;
    for (int v : arr) {
        cumulative_sum += v;
        if (cumulative_sum > 0) {
            result.push_back(cumulative_sum);
        }
    }
    return result;
}

// tests-start
int main() {
    vector<int> nums = {1, 2, 3, 4};
    auto result = getPositiveCumulativeSum(nums);
    for (int v : result) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 2 6
*/

File: public/dsa/arrays/positive_cumulative_sum/optimal/positive_cumulative_sum.py

from typing import List


class Solution:
    def getPositiveCumulativeSum(self, arr: List[int]) -> List[int]:
        result = []
        cumulative_sum = 0
        for v in arr:
            cumulative_sum += v
            if cumulative_sum > 0:
                result.append(cumulative_sum)
        return result


# tests-start
sol = Solution()
arr = [1, -2, 3, 4, -6]
sol.getPositiveCumulativeSum(arr)
# tests-end

'''output
[1, 2, 6]
'''
